### PR TITLE
Additional testing for many-to-many change tracking

### PR DESCRIPTION
**Pull request by ajcvickers**
Part of #19003

Mostly tests, but also fixes to navigation fixup to/from join entities

### CODE REVIEW

File: test/EFCore.Specification.Tests/Query/ManyToManyQueryFixtureBase.cs [hunk 1]
```diff
@@ -203,14 +203,13 @@ protected override void OnModelCreating(ModelBuilder modelBuilder, DbContext con
                     l => l.HasOne<EntityOne>().WithMany(e => e.JoinThreePayloadFullShared).HasForeignKey(e => e.OneId))
                 .HasKey(e => new { e.OneId, e.ThreeId });
 
-            // TODO: convert to shared type
```
Comment (smitpatel) [inline test/EFCore.Specification.Tests/Query/ManyToManyQueryFixtureBase.cs:206]: Leave this here, no? 😢

File: test/EFCore.SqlServer.FunctionalTests/ManyToManyTrackingSqlServerTest.cs [hunk 2]
```diff
@@ -21,6 +22,25 @@ protected override void UseTransaction(DatabaseFacade facade, IDbContextTransact
...
+
+                modelBuilder
+                    .Entity<JoinOneToThreePayloadFullShared>()
+                    .Property(e => e.Payload)
+                    .HasDefaultValue("Generated");
+
+                modelBuilder
+                    .Entity<JoinOneToThreePayloadFull>()
+                    .Property(e => e.Payload)
+                    .HasDefaultValue("Generated");            }
```
Comment (smitpatel) [inline test/EFCore.SqlServer.FunctionalTests/ManyToManyTrackingSqlServerTest.cs:43]: formatting

File: test/EFCore.Specification.Tests/Query/ManyToManyQueryFixtureBase.cs (hunk 1 above)
Comment (ajcvickers) [reply test/EFCore.Specification.Tests/Query/ManyToManyQueryFixtureBase.cs:206]: "SelfSkipPayloadLeft" doesn't have "Shared" in it's name...

Comment (ajcvickers) [reply test/EFCore.Specification.Tests/Query/ManyToManyQueryFixtureBase.cs:206]: Also the comment on the next line says "Join: Concrete".

**Pull request merged**